### PR TITLE
Add ASN.1 Integer comparison and to_owned

### PR DESCRIPTION
**Pull request by Skepfyr**
Comparing integers is actually infallible excitingly so we can implement the proper traits for it.

### CODE REVIEW

File: openssl/src/asn1.rs [hunk 1]
```diff
@@ -512,6 +511,23 @@ impl Asn1Integer {
...
 }
 
+impl Ord for Asn1Integer {
+    fn cmp(&self, other: &Self) -> Ordering {
+        Asn1IntegerRef::cmp(self, other)
+    }
+}
+impl PartialOrd for Asn1Integer {
+    fn partial_cmp(&self, other: &Asn1Integer) -> Option<Ordering> {
+        Some(self.cmp(other))
```
Comment (steffen-eiden) [inline openssl/src/asn1.rs:521]: Couldn't you just use `Asn1IntegerRef::partial_cmp` here as well?

Comment (Skepfyr) [reply openssl/src/asn1.rs:521]: I could, but I couldn't see it making any difference and I'd already written this code.

Comment (steffen-eiden) [reply openssl/src/asn1.rs:521]: So the stuff looks more similar.
yeah, its more a nit.

**Review by steffen-eiden — APPROVED** [review]

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**